junebug/service.py

```diff
@@ -0,0 +1,36 @@
+from twisted.application.service import MultiService
+from twisted.internet import reactor
+from twisted.internet.defer import inlineCallbacks
+from twisted.python import log
+from twisted.web.server import Site
+
+from junebug import JunebugApi
+
+
+class JunebugService(MultiService, object):
+    '''Base service that runs the HTTP API, and contains transports as child
+    services'''
+    def __init__(self, interface, port, redis_config, amqp_config):
+        super(JunebugService, self).__init__()
+        self.interface = interface
+        self.port = port
+        self.redis_config = redis_config
+        self.amqp_config = amqp_config
+
+    @inlineCallbacks
+    def startService(self):
+        '''Starts the HTTP server, and returns the port object that the server
+        is listening on'''
```

Comment:
Not too sure on this, but I think we might need to do the `super()` thing here and for `stopService()`.